### ISSUE TITLE
hsdev ignores cabal and using stack only, and makes error

### ISSUE THREAD

**Issue by QuietJoon** [SublimeHaskell/SublimeHaskell#392]
I think I configured SublimeHaskell to using cabal, not stack.
However, it seems to be that **SublimeHaskell does not tell hsdev not to using stack but to use cabal**.
Here is my user setting for SublimeHaskell.

```
{
    "add_to_PATH": ["/Users/yongjoon/Library/Haskell/bin/"],
    "haskell_build_tool": "cabal",
    "lint_check_fly_idle": 8
}
```
I'm sure that `/Users/yongjoon/Library/Haskell/bin/` includes `hsdev`, `cabal` and other required commands. And it is already included in `$PATH`.

After this setting, I've got SublimeHaskell error message like:
```
Uncategorized: 1

  <no file/command line/OPTIONS_GHC>: line 1, column 1:
    Uncategorized: Backend error encountered during 'Checking and Linting': ghc error
```

And this is sublime console errorlog:
https://pastebin.com/uXUrSTTT
```
<hsdev stderr> 2018-01-15 10:51:57 +0900    ERROR   :127.0.0.1:58905/update/task/project> Scope leaves with exception: tool '/usr/local/bin/stack' failed: exited with code 1: After installing Cabal, the package id couldn't be found (via ghc-pkg describe Cabal). This shouldn't happen, please report as a bug
<hsdev stderr>
<hsdev stderr> CallStack (from HasCallStack):
<hsdev stderr>   onError, called at src/System/Log/Simple/Monad.hs:104:36 in simple-log-0.9.3-LoOHmA7NyTW5wR9y7FxDp:System.Log.Simple.Monad
<hsdev stderr>
<hsdev stderr> 2018-01-15 10:51:57 +0900    ERROR   :127.0.0.1:58905/update/task> Scope leaves with exception: tool '/usr/local/bin/stack' failed: exited with code 1: After installing Cabal, the package id couldn't be found (via ghc-pkg describe Cabal). This shouldn't happen, please report as a bug
<hsdev stderr>
<hsdev stderr> CallStack (from HasCallStack):
<hsdev stderr>   onError, called at src/System/Log/Simple/Monad.hs:104:36 in simple-log-0.9.3-LoOHmA7NyTW5wR9y7FxDp:System.Log.Simple.Monad
<hsdev stderr>
<hsdev stderr> 2018-01-15 10:51:57 +0900	ERROR	:127.0.0.1:58905/update/task> Scope leaves with exception: tool '/usr/local/bin/stack' failed: exited with code 1: After installing Cabal, the package id couldn't be found (via ghc-pkg describe Cabal). This shouldn't happen, please report as a bug
<hsdev stderr> 

...
(repeated almost same error messages)
...

<hsdev stderr>
<hsdev stderr> 2018-01-15 10:51:59 +0900    ERROR   :127.0.0.1:58905/update> tool '/usr/local/bin/stack' failed: exited with code 1: After installing Cabal, the package id couldn't be found (via ghc-pkg describe Cabal). This shouldn't happen, please report as a bug
<hsdev stderr>
Sublime Haskell: complete returns error: tool error
msg: exited with code 1: After installing Cabal, the package id couldn't be found (via ghc-pkg describe Cabal). This shouldn't happen, please report as a bug
, id: 3, tool: /usr/local/bin/stack
```
I think this shows that SublimeHaskell is still not using cabal, but stack.
(And, it seems to be that there are more errors related to [hsdev](https://github.com/mvoidex/hsdev/issues/69).
I run `stack update` and I can sure Cabal library is installed, I confirmed it by `ghc-pkg describe Cabal`.
Anyway, this is not for SublimeHaskell issue. Please skip about the detail of log.)

To confirm this problem, I removed command `stack`, 
and I get another type of error log like 
https://pastebin.com/3FNknXKc
```
<hsdev stderr> 2018-01-15 10:45:10 +0900	ERROR	:127.0.0.1:58589/update/task/project> Scope leaves with exception: tool 'stack' not found
<hsdev stderr> CallStack (from HasCallStack):
<hsdev stderr>   onError, called at src/System/Log/Simple/Monad.hs:104:36 in simple-log-0.9.3-LoOHmA7NyTW5wR9y7FxDp:System.Log.Simple.Monad
<hsdev stderr> 
<hsdev stderr> 2018-01-15 10:45:10 +0900	ERROR	:127.0.0.1:58589/update/task> Scope leaves with exception: tool 'stack' not found
<hsdev stderr> CallStack (from HasCallStack):
<hsdev stderr>   onError, called at src/System/Log/Simple/Monad.hs:104:36 in simple-log-0.9.3-LoOHmA7NyTW5wR9y7FxDp:System.Log.Simple.Monad
<hsdev stderr> 
<hsdev stderr> 2018-01-15 10:45:10 +0900	ERROR	:127.0.0.1:58589/update> tool 'stack' not found
<hsdev stderr> 2018-01-15 10:45:10 +0900	ERROR	:127.0.0.1:58589/update/task/directory> Scope leaves with exception: tool 'stack' not found
....
(repeats)
....
<hsdev stderr> CallStack (from HasCallStack):
<hsdev stderr>   onError, called at src/System/Log/Simple/Monad.hs:104:36 in simple-log-0.9.3-LoOHmA7NyTW5wR9y7FxDp:System.Log.Simple.Monad
<hsdev stderr> 
<hsdev stderr> 2018-01-15 10:45:10 +0900	ERROR	:127.0.0.1:58589/update/task> Scope leaves with exception: tool 'stack' not found
<hsdev stderr> CallStack (from HasCallStack):
<hsdev stderr>   onError, called at src/System/Log/Simple/Monad.hs:104:36 in simple-log-0.9.3-LoOHmA7NyTW5wR9y7FxDp:System.Log.Simple.Monad
<hsdev stderr> 
<hsdev stderr> 2018-01-15 10:45:10 +0900	ERROR	:127.0.0.1:58589/update> tool 'stack' not found
Sublime Haskell: complete returns error: tool not found
```

Is stack indispensable command for SublimeHaskell?
I wish I misconfigured for using cabal of SublimeHaskell.

SublimText: Dev build 3156
GHC: 8.2.2
OS: macOS 10.13.2
SublimeHaskell: 2.1.16(not sure)~2.1.18

**Comment by bscottm** [comment]
@QuietJoon: _SublimeHaskell_ does not specifically depend on `stack`, but the `hsdev` backend does. `stack update` updates packages. You might need to do `stack upgrade` to upgrade `stack` itself.

How did you build `hsdev`?

**Comment by QuietJoon** [comment]
Just run `cabal install hsdev`.
When I run `stack upgrade`, I get the message like:
```
> stack upgrade
Current Stack version: 1.6.3, available download version: 1.6.3
Skipping binary upgrade, you are already running the most recent version
```
And `stack update` never helps the problem.

Anyway, I understand that this problem is not about SublimeHaskell. 
I didn't know that `hsdev` relies on `stack`.

Thanks for your comment.
